Library/Homebrew/config.rb

```diff
@@ -35,10 +35,8 @@
 # Must use /tmp instead of $TMPDIR because long paths break Unix domain sockets
 HOMEBREW_TEMP = Pathname.new(ENV.fetch("HOMEBREW_TEMP", "/tmp"))
 
-unless defined? HOMEBREW_LIBRARY_PATH
-  # Root of the Homebrew code base
-  HOMEBREW_LIBRARY_PATH = Pathname.new(__FILE__).realpath.parent
-end
+# Root of the Homebrew code base
+HOMEBREW_LIBRARY_PATH = Pathname.new(__FILE__).realpath.parent
```

Comment:
I don't think we can easily eliminate it, but I agree that the name can be confusing (and is known to have caused confusion in the past), thus we should try to find a better name.

We cannot eliminate it as easily because in regular execution `HOMEBREW_LIBRARY_PATH` is simply `HOMEBREW_LIBRARY/"Homebrew"`, thus we might be tempted to replace the former with the latter. However, in the testing code, they point to completely different locations in the file system and there is no obvious relationship between them. Here, `HOMEBREW_LIBRARY_PATH` is still the root of the Homebrew source code, but `HOMEBREW_LIBRARY` points to a temporary location and is not the parent directory of `HOMEBREW_LIBRARY_PATH`.

I'm happy to spend some time on this and to come up with a better name/solution, but I would prefer to do that in a separate follow-up PR. Is that fine by you?